bak2dvd/bak2dvd

```diff
@@ -0,0 +1,268 @@
+  latestVersion=$(httpGet https://api.github.com/repos/$githubUserName/$repositoryName/tags | grep -Eo '"name":.*?[^\\]",'| head -1 | grep -Eo "[0-9.]+" ) #always grabs the tag without the v option
+
+  if [[ $currentVersion == "" || $repositoryName == "" || $githubUserName == "" || $nameOfInstallFile == "" ]]; then
+    echo "Error: update utility has not been configured correctly." >&2
+    exit 1
+  elif [[ $latestVersion == "" ]]; then
+    echo "Error: no active internet connection" >&2
+    exit 1
+  else
+    if [[ "$latestVersion" != "$currentVersion" ]]; then
+      echo "Version $latestVersion available"
+      echo -n "Do you wish to update $repositoryName [Y/n]: "
+      read -r answer
+      if [[ "$answer" == [Yy] ]]; then
+        cd ~ || { echo 'Update Failed'; exit 1; }
+        if [[ -d  ~/$repositoryName ]]; then rm -r -f $repositoryName || { echo "Permissions Error: try running the update as sudo"; exit 1; } ; fi
+        git clone "https://github.com/$githubUserName/$repositoryName" || { echo "Couldn't download latest version"; exit 1; }
+        cd $repositoryName || { echo 'Update Failed'; exit 1; }
+        git checkout "v$latestVersion" 2> /dev/null || git checkout "$latestVersion" 2> /dev/null || echo "Couldn't git checkout to stable release, updating to latest commit."
+        chmod a+x install.sh #this might be necessary in your case but wasnt in mine.
+        ./$nameOfInstallFile "update" || exit 1
+        cd ..
+        rm -r -f $repositoryName || { echo "Permissions Error: update succesfull but cannot delete temp files located at ~/$repositoryName delete this directory with sudo"; exit 1; }
+      else
+        exit 1
+      fi
+    else
+      echo "$repositoryName is already the latest version"
+    fi
+  fi
+}
+
+checkInternet()
+{
+  httpGet github.com > /dev/null 2>&1 || { echo "Error: no active internet connection" >&2; return 1; } # query github with a get request
+}
+
+usage()
+{
+  cat <<EOF
+Bak2dvd
+Description: A script to write tar archives to optical disc.
+Usage: bak2dvd [flags] -t "[arguments]"
+  -u  Update Bash-Snippet Tools
+  -h  Show the help
+  -v  Get the tool version
+  -t  Parameters to pass to tar
+
+ Disc technology flags are:
+
+  -cd       for cd-rom"
+  -dvd      for single layer DVD"
+  -dvd-dl   for dual layer DVD"
+  -bd       for single layer BluRay"
+  -bd-dl    for dual layer BluRay"
+
+ Preset tar parameters are -cvpML and -f
+
+Example:
+   bak2dvd -d /dev/sr0 -dvd -t "-C $HOME Documents"
+   tool
+EOF
+}
+
+
+options=0
+
+while test $# -gt 0; do
+  case "$1" in
+    \?) echo "Invalid option: -$1" >&2
+        exit 1
+        ;;
+   -h|--help)  usage
+        exit 0
+        ;;
+    -v)  echo "Version $currentVersion"
+        exit 0
+        ;;
+    -u)  getConfiguredClient || exit 1
+        checkInternet || exit 1
+        update
+        exit 0
+        ;;
+    -d)
+	shift
+	if test $# -gt 0; then
+		export device=$1
```

Comment:
Any reason you are exporting all of these variables?

Reply:
Removed all the exports. Script works the same so they weren't needed.